pkg/controller/argocd/configmap_test.go

```diff
@@ -140,6 +143,51 @@ func TestReconcileArgoCD_reconcileArgoConfigMap_withDisableAdmin(t *testing.T) {
 	}
 }
 
+func TestReconcileArgoCD_reconcileArgoConfigMap_withDexConnector(t *testing.T) {
+	restoreEnv(t)
+	logf.SetLogger(logf.ZapLogger(true))
+	a := makeTestArgoCD(func(a *argoprojv1alpha1.ArgoCD) {
+		a.Spec.Dex.OpenShiftOAuth = true
+	})
+	sa := &corev1.ServiceAccount{
+		TypeMeta:   metav1.TypeMeta{Kind: "ServiceAccount", APIVersion: "v1"},
+		ObjectMeta: metav1.ObjectMeta{Name: "argocd-argocd-dex-server", Namespace: "argocd"},
+		Secrets: []corev1.ObjectReference{{
+			Name: "token",
+		}},
+	}
+
+	secret := argoutil.NewSecretWithName(metav1.ObjectMeta{Name: "token", Namespace: "argocd"}, "token")
+	r := makeTestReconciler(t, a, sa, secret)
+	err := r.reconcileArgoConfigMap(a)
+	assert.NilError(t, err)
+
+	cm := &corev1.ConfigMap{}
+	err = r.client.Get(context.TODO(), types.NamespacedName{
+		Name:      common.ArgoCDConfigMapName,
+		Namespace: testNamespace,
+	}, cm)
+	assert.NilError(t, err)
+
+	dex, ok := cm.Data["dex.config"]
+	if !ok {
+		t.Fatal("reconcileArgoConfigMap with dex failed")
+	}
+
+	m := make(map[string]interface{})
+	err = yaml.Unmarshal([]byte(dex), &m)
+	if err != nil {
+		t.Fatalf("unmarshal '%q' failed: %v", dex, err)
```

Comment:
Nit, you don't need `'%q'` here, `%q` will quote the value.

I suspect this could also be done with `assert.NilError` it has functionality for formatting a string.